test/image_test.py

```diff
@@ -508,22 +508,173 @@ def test_tostring_depth_24(self):
             'symmetric with "{}" format'.format(fmt),
         )
 
-    def todo_test_frombuffer(self):
+    def test_frombuffer(self):
+        """ test reading pixel data from a bytes buffer"""
+        pygame.display.init()
+        eight_bit_palette_buffer = bytearray([0, 0, 0, 0,
+                                              1, 1, 1, 1,
+                                              2, 2, 2, 2,
+                                              3, 3, 3, 3])
+
+        eight_bit_surf = pygame.image.frombuffer(eight_bit_palette_buffer,
+                                                 (4, 4), "P")
+        eight_bit_surf.set_palette([(255, 10, 20),
+                                    (255, 255, 255),
+                                    (0, 0, 0),
+                                    (50, 200, 20)])
+        self.assertEqual(eight_bit_surf.get_at((0, 0)),
+                         pygame.Color(255, 10, 20))
+        self.assertEqual(eight_bit_surf.get_at((1, 1)),
+                         pygame.Color(255, 255, 255))
+        self.assertEqual(eight_bit_surf.get_at((2, 2)),
+                         pygame.Color(0, 0, 0))
+        self.assertEqual(eight_bit_surf.get_at((3, 3)),
+                         pygame.Color(50, 200, 20))
+
+        rgb_buffer = bytearray([255, 10, 20,
```

Comment:
I guess these tests could be broken up into separate methods. The benefits being that you can run one at a time if debugging, and if one fails it's easier to see where the problem is.

Reply:
I have done this now.